Mage.Sets/src/mage/sets/mirrodin/RustElemental.java

```diff
@@ -0,0 +1,130 @@
+
+    static {
+        filter.add(new AnotherPredicate());
+        filter.add(new ControllerPredicate(TargetController.YOU));  
+    }
+
+    public RustElemental(UUID ownerId) {
+        super(ownerId, 234, "Rust Elemental", Rarity.UNCOMMON, new CardType[]{CardType.ARTIFACT, CardType.CREATURE}, "{4}");
+        this.expansionSetCode = "MRD";
+        this.subtype.add("Elemental");
+        this.power = new MageInt(4);
+        this.toughness = new MageInt(4);
+
+        // Flying
+        this.addAbility(FlyingAbility.getInstance());
+        // At the beginning of your upkeep, sacrifice an artifact other than Rust Elemental. If you can't, tap Rust Elemental and you lose 4 life.
+        this.addAbility(new OnEventTriggeredAbility(GameEvent.EventType.UPKEEP_STEP_PRE, "beginning of your upkeep", new RustElementalEffect(), false));
+    }
+
+    public RustElemental(final RustElemental card) {
+        super(card);
+    }
+       
+    @Override
+    public RustElemental copy() {
+        return new RustElemental(this);
+    }
+    
+    class RustElementalEffect extends OneShotEffect {
+        
+        public RustElementalEffect() {
+            super(Outcome.Damage);
+            this.staticText = "Sacrifice an artifact other than Rust Elemental. If you can't, tap Rust Elemental and you lose 4 life.";
+        }
+        
+        public RustElementalEffect(final RustElementalEffect effect) {
+            super(effect);
+        }
+        
+        @Override
+        public RustElementalEffect copy() {
+            return new RustElementalEffect(this);
+        }
+        
+        @Override
+        public boolean apply(Game game, Ability source) {
+            Permanent permanent = game.getPermanent(source.getSourceId());
+
+            if (permanent != null) {
+                // create cost for sacrificing an artifact
+                Player player = game.getPlayer(source.getControllerId());
+                if (player != null) {
+                    TargetControlledPermanent target = new TargetControlledPermanent(1, 1, filter, false);
+                    // if they can pay the cost, then they must pay
+                    if (target.canChoose(player.getId(), game)) {
+                        player.choose(Outcome.Sacrifice, target, source.getSourceId(), game);
+                        Permanent artifactSacrifice = game.getPermanent(target.getFirstTarget());
+                        if (artifactSacrifice != null) {
+                            // sacrifice the chosen artifact
+                            return artifactSacrifice.sacrifice(source.getSourceId(), game);
+                        }             
+                        else {
+                            permanent.tap(game);
+                            player.damage(4, source.getSourceId(), game, false, true);
+                        }
+                    }

```

Comment:
Yes. If you have no other artifacts to sacrifice, you have to tap Rust Elemental and you also lose 4 life.